plumgrid_playbooks/roles/plumgrid_director/tasks/main.yml

```diff
@@ -14,26 +14,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Extract License from file
 - name: Extract License
   set_fact:
     pg_license: "{{ lookup('file', pg_license_path) }}"
 
-# Modify template files
-- name: Setup Keepalived Config
+- name: Setup Keepalived and Nginx config
   template:
-    src: keepalived.conf
-    dest: /var/lib/libvirt/filesystems/plumgrid-data/conf/etc/keepalived.conf
-
-- name: Setup nginx Config
-  template:
-    src: default.conf
-    dest: /var/lib/libvirt/filesystems/plumgrid-data/conf/pg/nginx.conf
-
-- name: Restart PLUMgrid service
-  service:
-    name: plumgrid
-    state: restarted
+    src: "{{ item.src }}"
+    dest: "{{ plumgrid_data_dir }}/{{ item.dest }}"
+  with_items:
```

Comment:
@arbabnazar Yes i am testing it from scratch and here RESTART of service is certainly required. But handler does not restart service at this point, neither for directors nor for edges. Ultimately very next task (Login) for directors fails because it expects services to be started.
These three lines were just FYI. :)